kernel/src/expressions/scalars.rs

```diff
@@ -555,6 +556,22 @@ impl<T: Into<Scalar> + ToDataType> From<Option<T>> for Scalar {
     }
 }
 
+impl<K: Into<Scalar> + ToDataType, V: Into<Scalar> + ToDataType> From<HashMap<K, V>> for Scalar {
+    fn from(map: HashMap<K, V>) -> Self {
+        let map_type = MapType::new(K::to_data_type(), V::to_data_type(), false);
+
+        let pairs = map
+            .into_iter()
+            .map(|(k, v)| (k.into(), v.into()))
+            .collect::<Vec<_>>();
+
+        let map_data =
+            MapData::try_new(map_type, pairs).expect("Failed to create MapData from HashMap");
```

Comment:
ah unfortunately we basically need to make this a `TryFrom` instead of just a `From`. obviously this means we can't use the `derive(IntoEngineData)` - how about we just implement `IntoEngineData` ourselves for `CommitInfo` with a follow-up to expand the derive to leverage `TryFrom`?

Reply:
^ since we don't want to panic

Reply:
and can we make sure to test this?